redash/serializers/query_result.py

```diff
@@ -0,0 +1,92 @@
+import cStringIO
+import csv
+import xlsxwriter
+from dateutil.parser import parse as parse_date
+from redash.utils import json_loads, UnicodeWriter
+from redash.query_runner import (TYPE_BOOLEAN, TYPE_DATE, TYPE_DATETIME)
+from redash.authentication.org_resolving import current_org
+
+
+def convert_format(fmt):
+    return fmt.replace('DD', '%d').replace('MM', '%m').replace('YYYY', '%Y').replace('YY', '%y').replace('HH', '%H').replace('mm', '%M').replace('ss', '%s')
+
+
+def serialize_query_result_to_csv(query_result):
+    s = cStringIO.StringIO()
+
+    query_data = json_loads(query_result.data)
+
+    fieldnames = []
+    bool_columns = []

```

Comment:
This feels a tad too scripty for me. I'd prefer to have less details and noise here.
Perhaps this function could be split to several chunks?

1. Calling a function to slice `query_data['columns']` to three lists:
```python
bool_columns, date_columns, datetime_columns = split_columns_by_type(query_data['columns'])
```
2. Handling `csv.DictWriter` stuff and for each row:
2a. Calling a function to format booleans
3b. Calling a function to format dates
4c. Calling a function to format date times

Reply:
I've started with implementing 1 (https://github.com/getredash/redash/pull/3841/commits/5e0ccb992cba67e18c31c3ec34e08e1118feb99a), but then when I tried to split it further I realized I will need to pass around the lists 🤢or convert this into a class 🤔. So ended up doing this instead: https://github.com/getredash/redash/pull/3841/commits/feda3c9a163ce871e28918855fff4eaf912a6893.

I didn't do this initially as it felt too much, but it does clean the code and makes it easy to add support for new types later.